src/NHibernate/Dialect/Dialect.cs

```diff
@@ -2246,6 +2248,11 @@ protected void RegisterKeyword(string word)
 			Keywords.Add(word);
 		}
 
+		protected void RegisterKeywords(params string[] keywords)
+		{
+			Keywords.UnionWith(keywords);
```

Comment:
I do not know if this a really more performant overall, token lookups included, since an `Ordinal` comparison is maybe faster than an `OrdinalIgnoreCase`. But at least it is more robust, so I prefer having that ignore case.